target/linux/ramips/dts/mt7620a_apexera_netbot.dts

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "apexera,netbot", "ralink,mt7620a-soc";
+	model = "Apexera NetBot";
+
+	chosen {
+		bootargs = "console=ttyS1,57600";
+	};
+
+	aliases {
+		led-boot = &led_wifi;
+		led-failsafe = &led_wifi;
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		led_wifi: wifi {
+			label = "netbot:led:wifi";
+			gpios = <&gpio2 27 GPIO_ACTIVE_HIGH>;
+		};
+
+		wan {
+			label = "netbot:led:wan";
+			gpios = <&gpio2 26 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb2g {
+			label = "netbot:green:usb2";
+			gpios = <&gpio2 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb2r {
+			label = "netbot:red:usb2";
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb4g {
+			label = "netbot:green:usb4";
+			gpios = <&gpio2 3 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb4r {
+			label = "netbot:red:usb4";
+			gpios = <&gpio2 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb1g {
+			label = "netbot:green:usb1";
+			gpios = <&gpio2 21 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb1r {
+			label = "netbot:red:usb1";
+			gpios = <&gpio2 22 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb3g {
+			label = "netbot:green:usb3";
+			gpios = <&gpio0 22 GPIO_ACTIVE_HIGH>;
+		};
+
+		usb3r {
+			label = "netbot:red:usb3";
+			gpios = <&gpio0 23 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&i2s {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&i2c {
+	status = "okay";
+
+	pcf8563: rtc@51 {
+		status = "okay";
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
+
+	wm8978: wm8978@1a {
+		compatible = "wm8978";
+		reg = <0x1a>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
+		};
+	};
+};
+
+&sdhci {
+	status = "okay";
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ohci {
+	status = "okay";
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x10>;
```

Comment:
This can't be correct of you read the caldata from 0x0.

Please provide a list of MAC addresses on stock FW (2.4 Ghz, 5 Ghz, ethernet) and a hexdump of the factory partition. For further reading: https://openwrt.org/docs/guide-developer/mac.address